lib/pageflow/ability_mixin.rb

```diff
@@ -5,97 +5,200 @@ module AbilityMixin
+        AccountPolicy.new(user, account).read?
+      end
 
-      can [:read, :use_files], Entry, :id => user.entry_ids
+      can :update, Account do |account|
+        AccountPolicy.new(user, account).update?
+      end
 
-      can [:edit, :update, :edit_outline, :publish, :restore, :snapshot, :confirm_encoding], Entry do |entry|
-        can_edit_entry?(user, entry)
+      can :index, Account do |account|
+        AccountPolicy.new(user, account).index?
       end
 
-      can :manage, Storyline do |storyline|
-        can_edit_entry?(user, storyline.revision.entry)
+      can :see_link_to_index, Account, AccountPolicy::Scope.new(user, Account).resolve do |account|
+        AccountPolicy.new(user, account).see_link_to_index?
       end
 
-      can :manage, Chapter do |record|
-        can_edit_entry?(user, record.entry)
+      can :create, Membership do |membership|
+        membership.entity.nil? ||
+          membership.user.nil? ||
+          (!(membership.user.entries.include?(membership.entity) ||
+             membership.user.accounts.include?(membership.entity))) &&
+            MembershipPolicy.new(user, membership).create?
       end
 
-      can :manage, Pageflow.config.file_types.map(&:model) do |record|
-        can_edit_any_entry_using_file?(user, record)
+      can :index, Membership, MembershipPolicy::Scope.new(user, Membership).indexable
+
+      can :view, [Admin::MembersTab, Admin::RevisionsTab]
+
+      can :update, Membership do |membership|
+        MembershipPolicy.new(user, membership).edit_role?
       end
 
-      can :manage, Page do |page|
-        can_edit_entry?(user, page.chapter.entry)
+      can :destroy, Membership do |membership|
+        MembershipPolicy.new(user, membership).destroy?
       end
 
-      can :manage, Revision do |revision|
-        can_edit_entry?(user, revision.entry)
+      can :set_admin, ::User do |managed_user|
+        UserPolicy.new(user, managed_user).set_admin?
       end
 
-      can :view, [Admin::MembersTab, Admin::RevisionsTab]
+      can :see, :accounts do
+        user.admin? || user.memberships.on_accounts.length > 1
+      end
 
-      if user.admin?
-        can [:read, :create, :update], Account
-        can :destroy, Account do |account|
-          account.users.empty? && account.entries.empty?
+      unless user.admin?
+        can :configure_folder_on, Account do |account|
+          AccountPolicy.new(user, account).configure_folder_on?
         end
-        can :manage, Theming
 
-        can :manage, ::User
+        can :update_theming_on_entry_of, Account do |account|
+          AccountPolicy.new(user, account).update_theming_on_entry_of?
+        end
 
-        can :destroy, Membership
-        can :create, Membership do |membership|
-          membership.entry.nil? ||
-            membership.user.nil? ||
-            membership.entry.account == membership.user.account
+        if AccountPolicy::Scope.new(user, Account).entry_creatable.any?
```

Comment:
Without this check, we could get New Entry forms without Accounts to choose from (empty drop-down). Check moved to Entry Policy to increase consistency